tools/dpip/ipset.c

```diff
@@ -143,37 +196,52 @@ addr_arg_parse(char *arg, struct inet_addr_range *range, uint8_t *cidr)
     return EDPVS_OK;
 }
 
-/* [ tcp | udp: ]port1[-port2] */
-static void
+/* [ {tcp | udp | icmp | icmp6 }: ]port1[-port2] */
+static int
 port_arg_parse(char *arg, struct inet_addr_range *range)
 {
     char *proto = arg, *sep, *port1, *port2;
 
-    param.proto = IPPROTO_TCP;
+    if (!strncmp(proto, "tcp", 3))
+        param.proto = IPPROTO_TCP;
+    else if (!strncmp(proto, "udp", 3))
+        param.proto = IPPROTO_UDP;
+    else if (!strncmp(proto, "icmp", 4))
+        param.proto = IPPROTO_ICMP;
+    else if (!strncmp(proto, "icmp6", 5))
+        param.proto = IPPROTO_ICMPV6;
+    else
+        param.proto = 0;
+
     if ((sep = strchr(arg, ':')) != NULL) {
         *sep++ = '\0';
         arg = sep;
-        if (!strcmp(proto, "udp")) 
-            param.proto = IPPROTO_UDP;
     }
 
     port1 = arg;
     range->max_port = range->min_port = atoi(port1);
+    if (range->min_port > 65535)
```

Comment:
The type of min_port/max_port is __be16. Is this check necessary?

Reply:
fixed